src/core/mavsdk.h

```diff
@@ -131,14 +131,50 @@ class Mavsdk {
     ConnectionResult
     add_serial_connection(const std::string& dev_path, int baudrate = DEFAULT_SERIAL_BAUDRATE);
 
-    /**
-     * @brief Possible configurations.
-     */
-    enum class Configuration {
-        Autopilot, /**< @brief SDK is used as an autopilot. */
-        GroundStation, /**< @brief SDK is used as a ground station. */
-        CompanionComputer /**< @brief SDK is used on a companion computer onboard the system (e.g.
-                             drone). */
+    class Configuration {
```

Comment:
I'm not sure if we really need this configuration class. 

I'm wondering if it would be enough to have these setters:
```
Mavsdk::set_system_id(uint8_t system_id);
Mavsdk::set_component_id(uint8_t component_id);
Mavsdk::set_system_id_and_component_id(Configuration configuration);
```

@JonasVautherin what's your feel?

Reply:
@julianoes Actually @JonasVautherin suggested I use the class based method in the first place, rather than manually setting the ID via methods...

Reply:
I like the `Configuration` class better, but I'm coming from Java :see_no_evil:. In the end, it is a matter of preference, so I could live with both.

One thing is that I believe the `Configuration` class is more versatile: it could become a bigger thing (with more configuration options) without creating a mess. Whereas the 3 functions may have to completely change depending on how this all evolves (but it may not, that's a valid argument :laughing:).

Reply:
Ok convinced!